packages/apollo-cli/src/generate.ts

```diff
@@ -118,6 +115,52 @@ export default function generate(
         generatedFiles.map(o => o.content.fileContents).join("\n") + common
       );
 
+      writtenFiles += 1;
+    }
+  } else if (target === "typescript" || target === "ts") {
+    const context = compileToIR(schema, document, options);
+    const generatedFiles = generateTypescriptLocalSource(context);
+    const generatedGlobalFile = generateTypescriptGlobalSource(context);
+
+    const outFiles: {
+      [fileName: string]: BasicGeneratedFile;
+    } = {};
+
+    if (nextToSources || (fs.existsSync(outputPath) && fs.statSync(outputPath).isDirectory())) {
+      if (nextToSources && !fs.existsSync(outputPath)) {
+        fs.mkdirSync(outputPath);
```

Comment:
In this case you want to be checking and creating the joined path, right? I think you might need to move the `dir` adjustment logic from below up to the top of this branch so that you are using the same directory everywhere. Plus, it's expensive to check directories for every file so it would be faster to just do that once here.

Reply:
I am not sure I understand what you are trying to say. This line would create the directory for `globalTypes.ts` if we are placing next to sources. In that case we are not guaranteed to have the `/__generated__` (or whatever it is set to) directory (as there is no source next to it), so its checked here. The later `dir` logic is for the directories we need for placing next to the source. That part is just copied from the `flow` target. If instead we are using output flat, no checks or creates would happen except the initial one (on Line 129).

Reply:
Sorry, I was confusing `sourcePath` and `outputPath`. You're right, this makes sense for the `globalTypes.ts` file.